### PR TITLE
CLOUDP-169740 Use the official Community Server images

### PR DESCRIPTION
**Pull request by slaskawi**
### Summary

This Pull Request introduces the new official Community Server (https://hub.docker.com/r/mongodb/mongodb-community-server) images to the operator. The will become the default images from now on.

The Docker Inc. images might still be used but they require changing the MongoDB Image repo and image environment variables.

### All Submissions:

* [x] Have you opened an Issue before filing this PR?
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: docs/RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,20 +1,28 @@
-# MongoDB Kubernetes Operator 0.7.8
```
Comment (slaskawi) [inline docs/RELEASE_NOTES.md:1]: @dan-mckean Can I ask you to review this file?

File: docs/RELEASE_NOTES.md [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
+# MongoDB Kubernetes Operator 0.7.9
 
 ## Kubernetes Operator
 
 - Changes
-  - Fix a StatefulSet controller race condition with the mongodb-agent
-  - Fix merging label selectors when source is nil
-  - Improve logging verbosity
-  
+  - The Operator uses now the official [MongoDB Community Server images](https://hub.docker.com/r/mongodb/mongodb-community-server).
```
Comment (dan-mckean) [inline docs/RELEASE_NOTES.md:6]:
```suggestion
  - The Operator now uses the official [MongoDB Community Server images](https://hub.docker.com/r/mongodb/mongodb-community-server).
```

File: docs/RELEASE_NOTES.md [hunk 3]
```diff
@@ -1,20 +1,28 @@
...
+    ```asd
+          mongodb_image_name=mongo
+          mongodb_image_repo_url=docker.io
+    ```
+    Alternatively, it is possible to the Operator environmental variables to:
+    ```
+          MONGODB_IMAGE=mongo
+          MONGODB_REPO_URL=docker.io
+    ```
+    If the Operator is running using the default settings, the upgrade process will be automatic and seamless.
```
Comment (dan-mckean) [inline docs/RELEASE_NOTES.md:17]: How do I know if I'm not running the default settings? What settings?
And what do I do if I'm not?

File: test/e2e/feature_compatibility_version_upgrade/feature_compatibility_version_upgrade_test.go [hunk 4]
```diff
@@ -1,71 +0,0 @@
-package feature_compatibility_version_upgrade
```
Comment (mircea-cosbuc) [inline test/e2e/feature_compatibility_version_upgrade/feature_compatibility_version_upgrade_test.go:1]: How come we're getting rid of this test?

Comment (slaskawi) [reply test/e2e/feature_compatibility_version_upgrade/feature_compatibility_version_upgrade_test.go:1]: I squashed them with `feature_compatibility_version`. They were exactly the same copy of them apart from a single additional test. Now they all live in `feature_compatibility_version`.

File: docs/RELEASE_NOTES.md (hunk 2 above)
Comment (slaskawi) [reply docs/RELEASE_NOTES.md:6]: Fixed.

File: docs/RELEASE_NOTES.md (hunk 3 above)
Comment (slaskawi) [reply docs/RELEASE_NOTES.md]: @dan-mckean Added more text. Please check it again when you find a free slot.

Comment (dan-mckean) [reply docs/RELEASE_NOTES.md]: Great! Thanks you @slaskawi - fine to resolve!

**Review by dan-mckean — APPROVED** [review]

File: config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml [hunk 5]
```diff
@@ -5,12 +5,6 @@ kind: CustomResourceDefinition
 metadata:
   annotations:
     controller-gen.kubebuilder.io/version: v0.4.1
-    service.binding/type: 'mongodb'
```
Comment (nammn) [inline config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml:8]: I remember a discussion that we want to keep that in the crd base, albeit make removes it

File: controllers/construct/mongodbstatefulset.go [hunk 6]
```diff
@@ -40,6 +42,7 @@ const (
 	automationConfigEnv             = "AUTOMATION_CONFIG_MAP"
 	AgentImageEnv                   = "AGENT_IMAGE"
 	MongodbImageEnv                 = "MONGODB_IMAGE"
+	MongoDBImageType                = "IMAGE_TYPE"
```
Comment (nammn) [inline controllers/construct/mongodbstatefulset.go:45]: We had a discussion to make a new env's prefixed with `MDB_`: https://docs.google.com/document/d/1jWUQePeKPycmvOf-wx1mUuppdaw6eNOZqIMqHMzwaSA/edit?disco=AAAAtnwtYX0

File: controllers/construct/mongodbstatefulset.go [hunk 7]
```diff
@@ -307,11 +310,16 @@ func readinessProbeInit(volumeMount []corev1.VolumeMount) container.Modification
 
 func getMongoDBImage(version string) string {
 	repoUrl := os.Getenv(MongodbRepoUrl)
+	imageType := os.Getenv(MongoDBImageType)
```
Comment (nammn) [inline controllers/construct/mongodbstatefulset.go:313]: nit, can be in a follow-up:  wdyt about a helper function that returns the default if provided?
```
getEnvWithDefault(MongoDBImageType, DefaultImageType)
```

File: controllers/construct/mongodbstatefulset.go [hunk 8]
```diff
@@ -341,6 +349,7 @@ exec mongod -f %s;
 		container.WithImage(getMongoDBImage(version)),
 		container.WithResourceRequirements(resourcerequirements.Defaults()),
 		container.WithCommand(containerCommand),
+		container.WithArgs([]string{""}),
```
Comment (slaskawi) [inline controllers/construct/mongodbstatefulset.go:352]: Add a comment why

Comment (slaskawi) [reply controllers/construct/mongodbstatefulset.go:352]: done

File: controllers/construct/mongodbstatefulset.go (hunk 7 above)
Comment (slaskawi) [reply controllers/construct/mongodbstatefulset.go:313]: Done

File: controllers/construct/mongodbstatefulset.go (hunk 6 above)
Comment (slaskawi) [reply controllers/construct/mongodbstatefulset.go:45]: Done

File: config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml (hunk 5 above)
Comment (slaskawi) [reply config/crd/bases/mongodbcommunity.mongodb.com_mongodbcommunity.yaml:8]: I have added those to our types and upgraded the Controller Gen to pick them up.

File: docs/RELEASE_NOTES.md [hunk 9]
```diff
@@ -1,19 +1,33 @@
...
+# MongoDB Kubernetes Operator 0.8.0
 
 ## Kubernetes Operator
 
 - Changes
-  - Fixes a bug where the readiness probe would panic if the health-status file is missing.
-  - Make automation agent log level configurable.
+  - The Operator now uses the official [MongoDB Community Server images](https://hub.docker.com/r/mongodb/mongodb-community-server).
+    It is still possible to use the Docker Inc. images by altering the JSON configuration file:
+    ```asd
```
Comment (nammn) [inline docs/RELEASE_NOTES.md:8]:
```suggestion
    ```
```

**Review by nammn — APPROVED** [review]

File: docs/RELEASE_NOTES.md (hunk 9 above)
Comment (slaskawi) [reply docs/RELEASE_NOTES.md:8]: Cat on keyboard! :)

**Pull request merged**